shared/store/wallet.ts

```diff
@@ -60,226 +60,278 @@ const tokenDFI: WalletToken = {
   amount: '0',
   name: 'DeFiChain',
   displaySymbol: 'DFI (Token)',
-  avatarSymbol: 'DFI (Token)'
-}
+  avatarSymbol: 'DFI (Token)',
+};
 
 const utxoDFI: WalletToken = {
   ...tokenDFI,
   id: '0_utxo',
   displaySymbol: 'DFI (UTXO)',
-  avatarSymbol: 'DFI (UTXO)'
-}
+  avatarSymbol: 'DFI (UTXO)',
+};
 
 const unifiedDFI: WalletToken = {
   ...tokenDFI,
   id: '0_unified',
   displaySymbol: 'DFI',
-  avatarSymbol: 'DFI'
+  avatarSymbol: 'DFI',
+};
+
+async function getPaginatedResponse<T>(
+  api: (limit: number, next?: string) => Promise<ApiPagedResponse<T>>,
+): Promise<T[]> {
+  const current = [];
+  let hasNext = false;
+  let next;
+
+  try {
+    do {
+      // eslint-disable-next-line no-await-in-loop
+      const data: ApiPagedResponse<T> = await api(200, next);
+      current.push(...data);
+      hasNext = data.hasNext;
+      next = data.nextToken;
+    } while (hasNext);
+  } catch (e) {
+    return current;
+  }
+  return current;
 }
```

Comment:
There is a slightly simpler solution with `client.paginate`, if you're interested 😉 
```ts
  private async getAll<T>(method: () => Promise<ApiPagedResponse<T>>): Promise<T[]> {
    const batches = [await method()];
    while (batches[batches.length - 1].hasNext) {
      try {
        batches.push(await this.client.paginate(batches[batches.length - 1]));
      } catch (e) {
        break;
      }
    }

    return batches.reduce((prev, curr) => prev.concat(curr), [] as T[]);
  }
```

Reply:
yes thought of using our implementation too, but defaulted to LW implementation for now. We will anyway need to adapt those changes again in a greater scope, because there are other issues already, which just are really rare to occur